File: templates/matlab/joint_marginal.m
function fig = joint_marginal()
    addpath(fullfile(fileparts(mfilename('fullpath')),'..','..','_utils','matlab'));
    apply_theme(); rng(3); c = palette('cat',1);
    x = randn(600,1); y = 0.6*x + 0.7*randn(600,1);
    fig = figure('Position',[100 100 600 600]);
    ax_main  = subplot(4, 4, [5 6 7 9 10 11 13 14 15]);
    scatter(ax_main, x, y, 10, c, 'filled', 'MarkerFaceAlpha', 0.6);
    xlabel(ax_main, 'x'); ylabel(ax_main, 'y');
    ax_top   = subplot(4, 4, [1 2 3]);
    histogram(ax_top, x, 30, 'FaceColor', c, 'EdgeColor','w'); set(ax_top,'XTick',[]);
    ax_right = subplot(4, 4, [8 12 16]);
    histogram(ax_right, y, 30, 'FaceColor', c, 'EdgeColor','w', 'Orientation','horizontal');
    set(ax_right,'YTick',[]);
    sgtitle('Joint + marginals');
end

File: templates/python/joint_marginal.py
"""joint_marginal: 主散点 + 边缘直方图."""
import sys, os; sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', '_utils', 'python'))
import numpy as np
import matplotlib.pyplot as plt
from theme import apply_theme
from palette import cycle

def make_figure(x=None, y=None, title='Joint + marginals'):
    apply_theme(fig_size=(6, 6))
    if x is None:
        rng = np.random.default_rng(3)
        x = rng.normal(0, 1, 600); y = 0.6*x + rng.normal(0, 0.7, 600)
    fig = plt.figure()
    gs = fig.add_gridspec(4, 4, hspace=0.05, wspace=0.05)
    ax_main = fig.add_subplot(gs[1:, :-1])
    ax_top  = fig.add_subplot(gs[0, :-1], sharex=ax_main)
    ax_right= fig.add_subplot(gs[1:, -1], sharey=ax_main)
    ax_main.scatter(x, y, s=10, color=cycle(0), alpha=0.6, edgecolors='none')
    ax_top.hist(x,   bins=30, color=cycle(0), edgecolor='w')
    ax_right.hist(y, bins=30, color=cycle(0), edgecolor='w', orientation='horizontal')
    ax_top.tick_params(labelbottom=False)
    ax_right.tick_params(labelleft=False)
    ax_main.set_xlabel('x'); ax_main.set_ylabel('y')
    fig.suptitle(title)
    return fig

if __name__ == '__main__':
    fig = make_figure()
    fig.savefig(__file__.replace('.py', '.png'), dpi=150)
